### PR TITLE
[LINQ] Add example, and traits for filter, map, split

### PR DESCRIPTION
**Pull request by sguduguntla**
Here are just the Rust LINQ Operators

### CODE REVIEW

File: erdos/src/dataflow/operators/filter.rs
Comment (sukritkalra) [inline erdos/src/dataflow/operators/filter.rs:95]: Can you please add the name of the stream from `self` to the name of the operator? It might help us in debugging this at a future point.

File: erdos/src/dataflow/operators/map.rs
Comment (sukritkalra) [inline erdos/src/dataflow/operators/map.rs:92]: Same comment as for `FilterOperator`.

File: erdos/src/dataflow/operators/split.rs
Comment (sukritkalra) [inline erdos/src/dataflow/operators/split.rs:104]: Same comment as before.

File: examples/full_pipeline.rs
Comment (sukritkalra) [inline examples/full_pipeline.rs:420]: Can we have a separate example file for LINQ stuff?

**Review by sukritkalra — CHANGES_REQUESTED** [review]

File: examples/linq.rs
Comment (sukritkalra) [inline examples/linq.rs:35]: Please remove.

**Review by sukritkalra — APPROVED** [review]
LGTM with a minor change, but I'll wait for @pschafhalter to review before I resolve the merge conflicts.

File: erdos/src/dataflow/operators/filter.rs
Comment (pschafhalter) [inline erdos/src/dataflow/operators/filter.rs:81]:
```suggestion
    fn filter<F>(&self, filter_fn: F) -> Stream<D1>
       where F: 'static + Fn(&D1) -> bool + Send + Sync + Clone;
```

File: erdos/src/dataflow/operators/filter.rs
Comment (pschafhalter) [inline erdos/src/dataflow/operators/filter.rs:88]: Use where clause like above

File: erdos/src/dataflow/operators/map.rs
Comment (pschafhalter) [inline erdos/src/dataflow/operators/map.rs:80]: Use where clause.

File: erdos/src/dataflow/operators/map.rs
Comment (pschafhalter) [inline erdos/src/dataflow/operators/map.rs:88]: Use where clause.

File: examples/linq.rs
Comment (pschafhalter) [inline examples/linq.rs:74]: Remove state. Use `()` instead.

File: examples/linq.rs
Comment (pschafhalter) [inline examples/linq.rs:1]: I don't think we need this.

File: examples/linq.rs
Comment (pschafhalter) [inline examples/linq.rs:13]: Merge imports.

File: erdos/src/dataflow/operators/split.rs
Comment (pschafhalter) [inline erdos/src/dataflow/operators/split.rs:97]: Use where clause.

File: erdos/src/dataflow/operators/split.rs
Comment (pschafhalter) [inline erdos/src/dataflow/operators/split.rs:87]: Use where clause.

**Review by pschafhalter — CHANGES_REQUESTED** [review]

**Pull request merged**
